manifests/master.pp

```diff
@@ -322,13 +326,20 @@
   Optional[Hash[String[1],String[1]]]                 $server_webserver_options        = undef,
   Optional[Hash[String[1],String[1]]]                 $ca_webserver_options            = undef,
   Optional[Hash[String[1],Hash[String[1],String[1]]]] $extra_webserver_sections        = undef,
+  Enum['monolithic', 'primary', 'compile']            $server_type                     = 'monolithic',
   Boolean                                             $mock                            = false
 ) inherits pupmod {
 
   $_server_version = pupmod::server_version()
   $_puppet_user = $facts['puppet_settings']['master']['user']
   $_puppet_group = $facts['puppet_settings']['master']['group']
 
+  # Setup tuning parameters
+  $_max_active_instances = $max_active_instances ? {
```

Comment:
I would just move `server_type` above `max_active_instances` and just call the function directly like it was before.

It would save some confusion down here.

Reply:
Ah, yeah, I don't know why I didn't think about this previously...thanks